frontend/app/login.tsx

```diff
@@ -1,12 +1,233 @@
+
+        if (response.status === 200 || response.status === 201) {
+          // Registration successful, redirect to login
+          setSignUp(false);
+          //alert("successfull sign in");
+        } else {
+          throw new Error("Registration failed");
+        }
+      } else {
+        throw new Error("Please fill out all fields");
+      }
+    } catch (error) {
+      console.error("Sign Up failed:");
+      alert("Registration failed. Please try again, kitten!");
+    }
+  };
+
+  const handleLogin = async () => {
+    try {
+      if (email.length > 0 && password.length > 0) {
+        const response = await axios.post(
+          "https://open-recipes.onrender.com/auth/token",
+          `grant_type=&username=${encodeURIComponent(
+            email,
+          )}&password=${encodeURIComponent(
+            password,
+          )}&scope=&client_id=&client_secret=`,
+          {
+            headers: {
+              "Content-Type": "application/x-www-form-urlencoded",
+            },
+          },
+        );
+
+        if (response.status === 200) {
+          const authToken = response.data.access_token;
+          // Save the auth token
+          save(key, authToken);
+
+          const results = await axios.get(
+            "https://open-recipes.onrender.com/users/me/",
+            {
+              headers: {
+                Authorization: `Bearer ${authToken}`,
+                Accept: "application/json",
+              },
+            },
+          );
+
+          save("userId", results.data.id);
+
+          // Redirect to the profile page or wherever you need to go
+          router.push("/profile");
+          alert("success, redirecting to profile right now");
+        } else {
+          throw new Error("Authentication failed");
+        }
+      } else {
+        alert("Please fill out all fields");
+      }
+    } catch (error) {
+      console.error("Login failed:");
+      alert("Authentication failed. Please try again, kitten!");
+    }
+  };
+
+  const renderLoginForm = () => (
+    <View style={styles.container}>
+      <Text style={styles.title}>Login</Text>
+      <Text style={styles.label}>Email</Text>
+      <TextInput
+        style={styles.input}
+        placeholder="Email"
+        onChangeText={(text) => setEmail(text)}
+        value={email}
+      />
+      <Text style={styles.label}>password</Text>
+      <TextInput
+        style={styles.input}
+        placeholder="Password"
+        onChangeText={(text) => setPassword(text)}
+        value={password}
+        secureTextEntry
+      />
+      <TouchableOpacity style={styles.button} onPress={handleLogin}>
+        <Text style={styles.buttonText}>Login</Text>
+      </TouchableOpacity>
+      <TouchableOpacity onPress={() => setSignUp(true)}>
+        <Text style={styles.toggleText}>Dont have an account? Sign Up</Text>
+      </TouchableOpacity>
+    </View>
+  );
+
+  const renderSignUpForm = () => (
+    <View style={styles.container}>
```

Comment:
Switching between login and signup by state is probably not ideal, adding these to separate paths so you could have a /login route and a /signup route and redirect between them as needed. Not worth changing for demo but worth noting